data/common/src/main/java/com/example/data/common/api/di/NetworkModule.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.data.common.api.di
+
+import com.example.data.common.BuildConfig
+import com.example.data.common.api.CoinCapRetrofit
+import com.example.data.common.api.RetrofitBuilder
+import com.jakewharton.retrofit2.adapter.kotlin.coroutines.CoroutineCallAdapterFactory
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import java.util.concurrent.TimeUnit
+import javax.inject.Singleton
+
+/**
+ * @author yaya (@yahyalmh)
+ * @since 04th November 2022
+ */
+
+@Module
+@InstallIn(SingletonComponent::class)
+interface NetworkModule {
+    @Binds
+    fun bindCoincapRetrofit(coincapRetrofit: CoinCapRetrofit): RetrofitBuilder
+
+    companion object {
+        @Provides
+        fun provideHttpClient(): OkHttpClient = if (BuildConfig.DEBUG) {
+            val logging = HttpLoggingInterceptor()
+            logging.setLevel(HttpLoggingInterceptor.Level.BODY)
+            OkHttpClient.Builder()
+                .readTimeout(60, TimeUnit.SECONDS)
+                .writeTimeout(60, TimeUnit.SECONDS)
```

Comment:
It's a debatable issue - 60 seconds of general timeout seems huge, like 4-6 times too big - if you are blocking some action on screen during remote API call execution, then this mean you can block user screen for 60 seconds - it's better to fail sooner, i.e. after 15 seconds and then let user retry/go back to different screen/feature/etc.